### PR TITLE
Add chart for Fairwinds Insights

### PR DESCRIPTION
**Pull request by rbren**

### CODE REVIEW

**Comment by rbren** [comment]
I'd rather this chart _not_ build unless certain flags are specified (specifically, `organization`, `cluster`, and `base64token`). Is there a way to fix the tests without adding defaults in `values.yaml`?

**Review by bambash — COMMENTED** [review]
this looks great! 

I'm not entirely sure about testing, but you may be able to fix the CI process by following this pattern: https://github.com/FairwindsOps/charts/tree/master/incubator/letsencrypt-setup/ci

I noticed a few things that are statically defined which would be better served via templating.
* resources in `_job.spec` - it would be nice to override these via `values.yaml`
* non-templated `metadata.name` in all `rbac.yaml`
* non-templated `metadata.name` and `metadata.labels` in `serviceaccount.yaml`
* non-templated `metadata.name` in `token-secret.yaml`

It looks like a goldilocks chart tarball is pushed to `insights-agent/charts`, not sure if that is needed

**Comment by sudermanjr** [comment]
> certain flags are specified (specifically, `organization`, `cluster`, and `base64token`). Is there a way to fix the tests without adding defaults in `values.yaml`?

Whatever you specify in `ci/*-values.yaml` will be used during e2e testing. So you can tweak the tests that way.

File: scripts/ct.yaml
Comment (sudermanjr) [inline scripts/ct.yaml]: You're going to want to add stable here too. Goldilocks is moving to stable

File: stable/insights-agent/Chart.yaml
Comment (sudermanjr) [inline stable/insights-agent/Chart.yaml:5]: Curious why the chart is starting at 0.4.0 and the app at 0.0.1.  I would expect the opposite.

File: stable/insights-agent/Chart.yaml
Comment (sudermanjr) [inline stable/insights-agent/Chart.yaml]: Are you going to be the only maintainer?

File: stable/insights-agent/README.md
Comment (sudermanjr) [inline stable/insights-agent/README.md]: We generally try to stick to the `fairwinds-stable` convention due to our customers using incubator frequently.

File: stable/insights-agent/requirements.lock
Comment (sudermanjr) [inline stable/insights-agent/requirements.lock]: Once #145 gets merged, you will want to change this. My guess is it will merge before this one.

File: stable/insights-agent/requirements.yaml
Comment (sudermanjr) [inline stable/insights-agent/requirements.yaml]: I didn't know you could do this.  Neat.

File: stable/insights-agent/templates/token-secret.yaml
Comment (sudermanjr) [inline stable/insights-agent/templates/token-secret.yaml:2]: A lot of people (including Fairwinds) don't like create secrets using Helm due to how it stores those values in the cluster.  I would like to see this chart at least provide the option to specify a pre-existing secret name.

**Review by sudermanjr — CHANGES_REQUESTED** [review]
I agree with _Nate on the desire to see more templating around Rbac names and such. Also, I left some comments inline.

File: scripts/ct.yaml
Comment (rbren) [reply scripts/ct.yaml]: Done

File: stable/insights-agent/Chart.yaml
Comment (rbren) [reply stable/insights-agent/Chart.yaml:5]: This is a weird chart, since it involves several different "apps" (polaris, kube-hunter, goldilocks, ...).

I made `appVersion` `0.0.12` which matches the current version of the Insights application (which, ironically, is _not_ the application installed by this agent. but it's where the data is going, so...)

File: stable/insights-agent/Chart.yaml
Comment (rbren) [reply stable/insights-agent/Chart.yaml]: Added Barnabas

File: stable/insights-agent/README.md
Comment (rbren) [reply stable/insights-agent/README.md]: Done

File: stable/insights-agent/requirements.lock
Comment (rbren) [reply stable/insights-agent/requirements.lock]: Done

File: stable/insights-agent/requirements.yaml
Comment (rbren) [reply stable/insights-agent/requirements.yaml]: In fact, you cannot. :(

Moved to values.yaml

File: stable/insights-agent/templates/token-secret.yaml
Comment (rbren) [reply stable/insights-agent/templates/token-secret.yaml:2]: Makes sense. Done!

**Comment by rbren** [comment]
Think I addressed everything. Let me know if there's something I missed.

Thanks for all the feedback!

File: stable/insights-agent/Chart.yaml
Comment (sudermanjr) [reply stable/insights-agent/Chart.yaml:5]: ¯\(◉‿◉)/¯

File: stable/insights-agent/requirements.yaml
Comment (sudermanjr) [reply stable/insights-agent/requirements.yaml]: 😆

**Review by sudermanjr — COMMENTED** [review]
Looks like the build failed during docker hub outage. I am re-running

**Review by sudermanjr — APPROVED** [review]
Looks good! I'm sure there are bugs that got missed (it's a big chart), but good to go for this PR assuming the build passes this time.

**Pull request merged**
